ts/src/connections/sshHelpers.ts

```diff
@@ -136,43 +161,37 @@ export class SshHelpers {
             client.on('connect', (connection: any) => {
                 resolve(new ssh.WebSocketStream(new WebsocketStreamAdapter(connection)));
             });
-            client.on('connectFailed', (e: any) => {
-                if (e.message && e.message.startsWith('Error: ')) {
-                    e.message = e.message.substr(7);
+
+            // If the server responds but doesn't properly upgrade the connection to web socket, WebSocketClient fires 'httpResponse' event.
+            // TODO: Return ProblemDetails from TunnelRelay service
+            client.on('httpResponse', ({ statusCode, statusMessage }) => {
```

Comment:
Will calling `on('httpResponse')` throw an error in a browser environment where that event is not supported?

Reply:
In browser environment a different code branch uses a js web socket, not this custom node module.